### PR TITLE
Catch a JSON error and return undefined

### PR DESCRIPTION
**Pull request by mrashed-dev**
# Description
Sometimes the API won't respond with a JSON body, but given that the request completed without a status error, just return undefined.

# License
I confirm that this contribution is made under the terms of the MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: src/nylas-connection.ts [hunk 1]
```diff
@@ -309,7 +309,14 @@ export default class NylasConnection {
...
             } else {
-              return resolve(response.json());
+              return response
+                .json()
+                .then(json => {
+                  return resolve(json);
+                })
+                .catch(() => {
+                  return resolve(undefined);
+                });
```
Comment (AaronDDM) [inline src/nylas-connection.ts:319]: https://stevenklambert.com/writing/fetch-json-text-fallback/
This might not work!

Comment (mrashed-dev) [reply src/nylas-connection.ts:319]: Great find @AaronDDM, thank you! Fetch has been very frustrating to deal with regarding this.

Comment (james10424) [inline src/nylas-connection.ts:319]: nit:
```suggestion
                .then(resolve)
                .catch(() => resolve(undefined));
```

**Review by james10424 — APPROVED** [review]

File: src/nylas-connection.ts (hunk 1 above)
Comment (mrashed-dev) [reply src/nylas-connection.ts]: @AaronDDM after doing some research looks like the article's right. just added in the new clone method. Thanks for spotting this!

**Review by AaronDDM — APPROVED** [review]
🎉

**Pull request merged**